src/bindings/python/src/pyopenvino/graph/dimension.cpp

```diff
@@ -121,6 +121,34 @@ void regclass_graph_Dimension(py::module m) {
                 :return: Value of the dimension.
                 :rtype: int
             )");
+
+    /// Symbol-related methods: START
+    dim.def("has_symbol",
+            &ov::Dimension::has_symbol,
+            R"(
+              Check if Dimension has meaningful symbol.
+              :return: True if symbol was set, else False.
+              :rtype: bool
+            )");
+    dim.def("get_symbol",
+            &ov::Dimension::get_symbol,
+            R"(
+                Return this dimension's symbol as Symbol object.
+
+                :return: Value of the dimension.
+                :rtype: openvino.Symbol
+            )");
+
+    dim.def("set_symbol",
+            &ov::Dimension::set_symbol,
+            py::arg("symbol"),
+            R"(
+                Sets provided Symbol as this dimension's symbol.
+                :param symbol: The symbol to set to this dimension.
+                :type symbol: openvino.Symbol
+            )");
+    /// Symbol-related methods: END
```

Comment:
I suggest also to add property `symbol`
@jiwaszki what do you think?

Reply:
Will it allow to get and set the value through it? If so, I would need an example of how to set it up.

Reply:
Let's not duplicate/simplify API for now. As now it is aligned, it's good enough.

Reply:
yep,
please use [this ](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/pyopenvino/graph/descriptors/tensor.cpp#L124) as example

Reply:
got it. then no simplifications for now, will think about it in the future.